horovod/horovod.sh

```diff
@@ -0,0 +1,129 @@
+TENSORFLOW_VERSION="$(/usr/share/google/get_metadata_value attributes/tensorflow-version || echo ${DEFAULT_TENSORFLOW_VERSION})"
+readonly TENSORFLOW_VERSION
+PYTORCH_VERSION="$(/usr/share/google/get_metadata_value attributes/pytorch-version || echo ${DEFAULT_PYTORCH_VERSION})"
+readonly PYTORCH_VERSION
+TORCHVISION_VERSION="$(/usr/share/google/get_metadata_value attributes/torvision-version || echo ${DEFAULT_TORCHVISION_VERSION})"
+readonly TORCHVISION_VERSION
+MXNET_VERSION="$(/usr/share/google/get_metadata_value attributes/mxnet-version || echo ${DEFAULT_MXNET_VERSION})"
+readonly MXNET_VERSION
+
+CUDA_VERSION="$(/usr/share/google/get_metadata_value attributes/cuda-version || echo ${DEFAULT_CUDA_VERSION})"
+readonly CUDA_VERSION
+
+HOROVOD_ENV_VARS="$(/usr/share/google/get_metadata_value attributes/horovod-env-vars || echo "")"
+readonly HOROVOD_ENV_VARS
+
+INSTALL_MPI="$(/usr/share/google/get_metadata_value attributes/install-mpi || echo "")"
+readonly INSTALL_MPI
+
+function install_mpi() {
+  local -r MPI_VERSION="4.1.0"
+  local -r MPI_URL="https://download.open-mpi.org/release/open-mpi/v4.1/openmpi-${MPI_VERSION}.tar.gz"
+  #apt install openmpi
+  
+  tmp_dir=$(mktemp -d -t mlvm-horovod-mpi-XXXX)
+  wget -nv --timeout=30 --tries=5 --retry-connrefused -P ${tmp_dir} "${MPI_URL}" 
+  gunzip -c "${tmp_dir}/openmpi-${MPI_VERSION}.tar.gz" | tar xf - -C ${tmp_dir}
+
+  cur_dir=$(pwd)  
+  cd "${tmp_dir}/openmpi-${MPI_VERSION}"
+  ./configure --prefix=/usr/local --enable-mpirun-prefix-by-default
+  make all install
+  ldconfig
+  cd ${cur_dir}
+}
+
+function configure_hadoop_env() {
+  # Horovod relies on CLASSPATH being set in the environment. This needs to
+  # contain the Hadoop jars.
+  cat << 'EOF' >> /etc/profile.d/horovod.sh
+export CLASSPATH=$(hadoop classpath --glob)
+EOF
+}
+
+function install_frameworks() {
+  local frameworks=()
+
+  # Add gpu-versions of libraries
+  if (lspci | grep -q NVIDIA); then
+    pip install "torch==${PYTORCH_VERSION}+cu${CUDA_VERSION//.}" "torchvision==${TORCHVISION_VERSION}+cu${CUDA_VERSION//.}" -f "https://download.pytorch.org/whl/torch_stable.html"
+    if [[ ${TENSORFLOW_VERSION} == "1."* ]]; then
+      frameworks+=("tensorflow-gpu==${TENSORFLOW_VERSION}")
+    else
+      frameworks+=("tensorflow==${TENSORFLOW_VERSION}")
+    fi
+  else
+    frameworks+=(
+      "torch==${PYTORCH_VERSION}" 
+      "torchvision==${TORCHVISION_VERSION}"
+      "tensorflow==${TENSORFLOW_VERSION}"
+      "mxnet==${MXNET_VERSION}"
+      )
+  fi
+
+  pip install "${frameworks[@]}"
+}
+
+function install_horovod() {
+  # Install cmake
+  apt-get install -y cmake
+
+  # Install Horovod
+  horovod_build_frameworks="tensorflow,pytorch,spark"
+  
+  # Horovod on Dataproc with MXNet only supported with CPUs
+  if ! (lspci | grep -q NVIDIA); then
+    horovod_build_frameworks+=",mxnet"
+  fi
+
+  eval "${HOROVOD_ENV_VARS} pip install --no-cache-dir horovod[${horovod_build_frameworks}]==${HOROVOD_VERSION}"
```

Comment:
Use `conda` if possible

Reply:
There's no reliable release of Horovod on any Conda channel